<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Examples for installation checking &mdash; pyVIA  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="1. Basic workflow" href="Basic%20tutorial.html" />
    <link rel="prev" title="Parameters and Attributes" href="Parameters%20and%20Attributes.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> pyVIA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">General:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Parameters%20and%20Attributes.html">Parameters and Attributes</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Examples for installation checking</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#a-b-toy-data-multifurcation-and-disconnected"><strong>1.a/b Toy data (Multifurcation and Disconnected)</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#a-general-input-format-and-wrapper-function-uses-example-of-pre-b-cell-differentiation"><strong>2.a General input format and wrapper function (uses example of pre-B cell differentiation)</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#b-via-wrapper-for-generic-disconnected-trajectory"><strong>2.b VIA wrapper for generic disconnected trajectory</strong></a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">Tutorials:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Basic%20tutorial.html">1. Basic workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="Disconnected%20trajectories.html">2.Disconnected Trajectories</a></li>
<li class="toctree-l1"><a class="reference internal" href="Hematopoiesis-scRNAseq.html">3. Human Hematopoiesis (scRNAseq)</a></li>
<li class="toctree-l1"><a class="reference internal" href="Imaging%20Cytometry%20%28cell%20cycle%29.html">4. Imaging cytometry</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">pyVIA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Examples for installation checking</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/Basic Example Code.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="examples-for-installation-checking">
<h1>Examples for installation checking<a class="headerlink" href="#examples-for-installation-checking" title="Permalink to this headline"></a></h1>
<p>The examples below show how to run VIA on generic connected and disconnected data using wrapper functions and serve as a check for your installation. For more detailed guidelines on running VIA and plotting the results, please use the Notebooks. We also highlight a few difference in calling VIA when using Windows versus Linux. The data for the Jupyter Notebooks and Examples are available in the <a class="reference external" href="https://github.com/ShobiStassen/VIA/tree/master/Datasets">Datasets folder</a> (smaller files) with larger datasets <a class="reference external" href="https://drive.google.com/drive/folders/1WQSZeNixUAB1Sm0Xf68ZnSLQXyep936l?usp=sharing">here</a></p>
<p>A <a class="reference external" href="https://github.com/ShobiStassen/VIA/blob/master/test_pyVIA.py">test script</a> is available for some of the different datasets, please change the foldername accordingly to the folder containing relevant data files</p>
<ul class="simple">
<li><p>1.a Toy Data (multifurcation) <a class="reference external" href="https://github.com/ShobiStassen/VIA/blob/master/Jupyter%20Notebooks/ViaJupyter_Toy_Multifurcating.ipynb">Multifurcation Jupyter NB</a></p></li>
<li><p>1.b Toy Data (disconnected) <a class="reference external" href="https://github.com/ShobiStassen/VIA/blob/master/Jupyter%20Notebooks/ViaJupyter_Toy_Disconnected.ipynb">Disconnected Jupyter NB</a></p></li>
<li><p>2.a <strong>General input data</strong> formatting and wrapper function</p></li>
<li><p>2.b <strong>General disconnected trajectories</strong> wrapper function</p></li>
<li><p>3.a Human Embryoid Bodies (wrapper function for testing VIA)</p></li>
<li><p>3.b Human Embryoid Bodies (Configuring VIA) <a class="reference external" href="https://github.com/ShobiStassen/VIA/blob/master/Jupyter%20Notebooks/ViaJupyter_EmbryoidBody.ipynb">EB Jupyter NB</a></p></li>
</ul>
<div class="section" id="a-b-toy-data-multifurcation-and-disconnected">
<h2><strong>1.a/b Toy data (Multifurcation and Disconnected)</strong><a class="headerlink" href="#a-b-toy-data-multifurcation-and-disconnected" title="Permalink to this headline"></a></h2>
<p>Two examples <a class="reference external" href="https://github.com/ShobiStassen/VIA/tree/master/Datasets">toy datasets</a>  with annotations generated using DynToy are provided. For the step-by-step code within these wrappers, please see the corresponding Jupyter NBs.</p>
<p><strong>1.a/b Run on Linux</strong></p>
<p>All examples are shown according to Linux OS, small modifications are required to run on a Windows OS (see below):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pyVIA.core</span> <span class="k">as</span> <span class="nn">via</span>
<span class="c1"># ensure the data and label files are in csv format when you download/save them</span>
<span class="c1"># multifurcation</span>
<span class="c1"># the root is automatically set to  root_user = &#39;M1&#39;</span>
<span class="n">via</span><span class="o">.</span><span class="n">main_Toy</span><span class="p">(</span><span class="n">ncomps</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span><span class="n">dataset</span><span class="o">=</span><span class="s1">&#39;Toy3&#39;</span><span class="p">,</span> <span class="n">random_seed</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span><span class="n">foldername</span> <span class="o">=</span> <span class="s2">&quot;.../Trajectory/Datasets/&quot;</span><span class="p">)</span> <span class="c1">#multifurcation</span>
<span class="c1"># disconnected trajectory</span>
<span class="c1"># In the wrapper for Toy, the root is automatically set as a list root_user = [&#39;T1_M1&#39;, &#39;T2_M1&#39;] # e.g. T2_M3 is a cell belonging to the 3rd Milestone (M3) of the   second Trajectory (T2)</span>
<span class="n">via</span><span class="o">.</span><span class="n">main_Toy</span><span class="p">(</span><span class="n">ncomps</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span><span class="n">dataset</span><span class="o">=</span><span class="s1">&#39;Toy4&#39;</span><span class="p">,</span><span class="n">random_seed</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span><span class="n">foldername</span> <span class="o">=</span><span class="s2">&quot;.../Trajectory/Datasets/&quot;</span><span class="p">)</span> <span class="c1">#2 disconnected trajectories</span>
</pre></div>
</div>
<p><strong>1.a/b Run on Windows</strong></p>
<p>Windows may require minor modifications in calling the code due to the way multiprocessing works in Windows compared to Linux:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">#when running from an IDE you need to call the function in the following way to ensure the parallel processing works:</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pyVIA.core</span> <span class="k">as</span> <span class="nn">via</span>
<span class="n">f</span><span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="sa">r</span><span class="s1">&#39;C:\Users\...\Documents&#39;</span><span class="o">+</span><span class="s1">&#39;</span><span class="se">\\</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="k">def</span> <span class="nf">main</span><span class="p">():</span>
    <span class="n">via</span><span class="o">.</span><span class="n">main_Toy</span><span class="p">(</span><span class="n">ncomps</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span><span class="n">dataset</span><span class="o">=</span><span class="s1">&#39;Toy3&#39;</span><span class="p">,</span> <span class="n">random_seed</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span><span class="n">foldername</span><span class="o">=</span> <span class="n">f</span><span class="p">)</span>
<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span><span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="n">main</span><span class="p">()</span>

<span class="c1">#when running directly from terminal:</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pyVIA.core</span> <span class="k">as</span> <span class="nn">via</span>
<span class="n">f</span><span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="sa">r</span><span class="s1">&#39;C:\Users\...\Documents&#39;</span><span class="o">+</span><span class="s1">&#39;</span><span class="se">\\</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">via</span><span class="o">.</span><span class="n">main_Toy</span><span class="p">(</span><span class="n">ncomps</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span><span class="n">dataset</span><span class="o">=</span><span class="s1">&#39;Toy3&#39;</span><span class="p">,</span> <span class="n">random_seed</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span><span class="n">foldername</span><span class="o">=</span> <span class="n">f</span><span class="p">)</span>
</pre></div>
</div>
<p><a href="#id1"><span class="problematic" id="id2">**</span></a>Multifurcating toy dataset 1.a ** <a class="reference external" href="https://shobistassen.github.io/bifurc.html">*(click to open interactive graph)*</a></p>
<img src= "https://github.com/ShobiStassen/VIA/blob/master/Figures/Toy3_MainFig.png?raw=true" width="600px" align="center" </a>

<img src= "https://github.com/ShobiStassen/VIA/blob/master/Figures/streamplotspng.png?raw=true" width="600px" align="center" </a><p><strong>Disconnected toy dataset 1.b</strong> <a class="reference external" href="https://shobistassen.github.io/bifurc.html">(click to open interactive graph)</a></p>
<img src="https://github.com/ShobiStassen/VIA/blob/master/Figures/Toy4_MainFig.png?raw=true" width="600px" align="center" </a></div>
<div class="section" id="a-general-input-format-and-wrapper-function-uses-example-of-pre-b-cell-differentiation">
<h2><strong>2.a General input format and wrapper function (uses example of pre-B cell differentiation)</strong><a class="headerlink" href="#a-general-input-format-and-wrapper-function-uses-example-of-pre-b-cell-differentiation" title="Permalink to this headline"></a></h2>
<p>These wrapper functions are a good start but we highly recommend you look at the tutorials as you will be afforded a much higher degree of control without much added complexity. The below wrappers operate in the 2-iteration format (a coarse followed by a fine-grained), but this is not always needed and you will have more intuitive for the behaviour of your data by following the steps in the Tutorials. Nonetheless, the following wrappers are a great way to start to familiarize yourself with the various outputs from VIA.</p>
<p>Datasets and labels used in this example are provided in <a class="reference external" href="https://github.com/ShobiStassen/VIA/tree/master/Datasets">Datasets</a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Read the two files:</span>
<span class="c1"># 1) The first file contains 200PCs of the Bcell filtered and normalized data for the first 5000 HVG.</span>
<span class="c1"># 2) The second file contains raw count data for marker genes</span>

<span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;./Bcell_200PCs.csv&#39;</span><span class="p">)</span>
<span class="n">data_genes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;./Bcell_markergenes.csv&#39;</span><span class="p">)</span>
<span class="n">data_genes</span> <span class="o">=</span> <span class="n">data_genes</span><span class="o">.</span><span class="n">drop</span><span class="p">([</span><span class="s1">&#39;cell&#39;</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">true_label</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;time_hour&#39;</span><span class="p">]</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">drop</span><span class="p">([</span><span class="s1">&#39;cell&#39;</span><span class="p">,</span> <span class="s1">&#39;time_hour&#39;</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">AnnData</span><span class="p">(</span><span class="n">data_genes</span><span class="p">)</span>
<span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;X_pca&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">values</span>

<span class="c1"># use UMAP or PHate to obtain embedding that is used for single-cell level visualization</span>
<span class="n">embedding</span> <span class="o">=</span> <span class="n">umap</span><span class="o">.</span><span class="n">UMAP</span><span class="p">(</span><span class="n">random_state</span><span class="o">=</span><span class="mi">42</span><span class="p">,</span> <span class="n">n_neighbors</span><span class="o">=</span><span class="mi">15</span><span class="p">,</span> <span class="n">init</span><span class="o">=</span><span class="s1">&#39;random&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">values</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">])</span>

<span class="c1"># list marker genes or genes of interest if known in advance. otherwise marker_genes = []</span>
<span class="n">marker_genes</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Igll1&#39;</span><span class="p">,</span> <span class="s1">&#39;Myc&#39;</span><span class="p">,</span> <span class="s1">&#39;Slc7a5&#39;</span><span class="p">,</span> <span class="s1">&#39;Ldha&#39;</span><span class="p">,</span> <span class="s1">&#39;Foxo1&#39;</span><span class="p">,</span> <span class="s1">&#39;Lig4&#39;</span><span class="p">,</span> <span class="s1">&#39;Sp7&#39;</span><span class="p">]</span>  <span class="c1"># irf4 down-up</span>
<span class="c1"># call VIA. We identify an early (suitable) start cell root = [42]. Can also set an arbitrary value</span>
<span class="n">via</span><span class="o">.</span><span class="n">via_wrapper</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">true_label</span><span class="p">,</span> <span class="n">embedding</span><span class="p">,</span> <span class="n">knn</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">ncomps</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">jac_std_global</span><span class="o">=</span><span class="mf">0.15</span><span class="p">,</span> <span class="n">root</span><span class="o">=</span><span class="p">[</span><span class="mi">42</span><span class="p">],</span> <span class="n">dataset</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">,</span>
            <span class="n">random_seed</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="n">v0_toobig</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span> <span class="n">v1_toobig</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span> <span class="n">marker_genes</span><span class="o">=</span><span class="n">marker_genes</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="b-via-wrapper-for-generic-disconnected-trajectory">
<h2><strong>2.b VIA wrapper for generic disconnected trajectory</strong><a class="headerlink" href="#b-via-wrapper-for-generic-disconnected-trajectory" title="Permalink to this headline"></a></h2>
<p>A slightly different wrapper is called for the disconnected scenario. Refer to the Jupytern NB for a step-by-step tutorial.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="c1">#foldername corresponds to the location where you have saved the Toy Disconnected data (shown in example 2)</span>
<span class="c1">#Read in the data and labels</span>
<span class="n">df_counts</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">foldername</span> <span class="o">+</span> <span class="s2">&quot;toy_disconnected_M9_n1000d1000.csv&quot;</span><span class="p">,</span> <span class="s1">&#39;rt&#39;</span><span class="p">,</span> <span class="n">delimiter</span><span class="o">=</span><span class="s2">&quot;,&quot;</span><span class="p">)</span>
<span class="n">df_ids</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">foldername</span> <span class="o">+</span> <span class="s2">&quot;toy_disconnected_M9_n1000d1000_ids.csv&quot;</span><span class="p">,</span> <span class="s1">&#39;rt&#39;</span><span class="p">,</span> <span class="n">delimiter</span><span class="o">=</span><span class="s2">&quot;,&quot;</span><span class="p">)</span>

<span class="c1"># Make AnnData object for wrapper function to read-in data and do PCA</span>
<span class="n">df_ids</span><span class="p">[</span><span class="s1">&#39;cell_id_num&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="nb">int</span><span class="p">(</span><span class="n">s</span><span class="p">[</span><span class="mi">1</span><span class="p">::])</span> <span class="k">for</span> <span class="n">s</span> <span class="ow">in</span> <span class="n">df_ids</span><span class="p">[</span><span class="s1">&#39;cell_id&#39;</span><span class="p">]]</span>
<span class="n">df_counts</span> <span class="o">=</span> <span class="n">df_counts</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="s1">&#39;Unnamed: 0&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
<span class="n">df_ids</span> <span class="o">=</span> <span class="n">df_ids</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;cell_id_num&#39;</span><span class="p">])</span>
<span class="n">df_ids</span> <span class="o">=</span> <span class="n">df_ids</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">true_label</span> <span class="o">=</span> <span class="n">df_ids</span><span class="p">[</span><span class="s1">&#39;group_id&#39;</span><span class="p">]</span>
<span class="n">adata_counts</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">AnnData</span><span class="p">(</span><span class="n">df_counts</span><span class="p">,</span> <span class="n">obs</span><span class="o">=</span><span class="n">df_ids</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata_counts</span><span class="p">,</span> <span class="n">svd_solver</span><span class="o">=</span><span class="s1">&#39;arpack&#39;</span><span class="p">,</span> <span class="n">n_comps</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="c1">#Since there are 2 disconnected trajectories, we provide 2 arbitrary roots (start cells).If there are more disconnected paths, then VIA arbitrarily selects roots. #The root can also just be arbitrarily set as [1] and VIA can detect how many additional roots it must add</span>
<span class="c1"># The root can also be provided as a cell type level label corresponding to the groups present in &quot;true_label&quot;, in this case the dataset must be set to &#39;group&#39;</span>
<span class="n">via</span><span class="o">.</span><span class="n">via_wrapper_disconnected</span><span class="p">(</span><span class="n">adata_counts</span><span class="p">,</span> <span class="n">true_label</span><span class="p">,</span> <span class="n">embedding</span><span class="o">=</span><span class="n">adata_counts</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;X_pca&#39;</span><span class="p">][:,</span> <span class="mi">0</span><span class="p">:</span><span class="mi">2</span><span class="p">],</span> <span class="n">root</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span> <span class="n">preserve_disconnected</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span> <span class="n">ncomps</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span><span class="n">cluster_graph_pruning_std</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>

<span class="c1">#in the case of connected data (i.e. only 1 graph component. e.g. Toy Data Multifurcating) then the wrapper function from example 3.a can be used:</span>
<span class="n">via</span><span class="o">.</span><span class="n">via_wrapper</span><span class="p">(</span><span class="n">adata_counts</span><span class="p">,</span> <span class="n">true_label</span><span class="p">,</span> <span class="n">embedding</span><span class="o">=</span>  <span class="n">adata_counts</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;X_pca&#39;</span><span class="p">][:,</span><span class="mi">0</span><span class="p">:</span><span class="mi">2</span><span class="p">],</span> <span class="n">root</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span> <span class="n">ncomps</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span><span class="n">cluster_graph_pruning_std</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="Parameters%20and%20Attributes.html" class="btn btn-neutral float-left" title="Parameters and Attributes" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="Basic%20tutorial.html" class="btn btn-neutral float-right" title="1. Basic workflow" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Shobana Stassen.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>